### DISCUSSION TITLE
FastAPI 0.113.0 does not support using form models together with UploadFile

### DISCUSSION THREAD

**Discussion by tosmart01** [fastapi/fastapi#12138] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import BaseModel, Field
class ModelInfoBase(BaseModel):
    model_name: str = Field(..., max_length=255)
    api_url: str = Field(..., max_length=255)

async def post(self, data: Annotated[ModelInfoCreate, Form()],
                   image: Annotated[UploadFile, File()] = None,
                   ):
    ...
```

### Description

![image](https://github.com/user-attachments/assets/c31346ae-9e30-4f96-ac58-3a1f52e17137)

### Operating System

macOS

### FastAPI Version

0.113.0

### Pydantic Version

2.8.0

### Python Version

3.11.4

**Comment by pythonweb2** [comment]
It would help to make a reproducible example, the code you posted is incomplete

**Reply by pythonweb2** [reply]
I fixed the code, in this case, just put the file inside of the model.

```python
from typing import Annotated

from fastapi import FastAPI, File, Form, UploadFile

app = FastAPI()


from pydantic import BaseModel, Field


class Info(BaseModel):
    name: str = Field(..., max_length=255)
    api_url: str = Field(..., max_length=255)
    image: Annotated[UploadFile, File()] = None


@app.post("/login/")
async def login(
    data: Annotated[Info, Form()],
):
    return data
```

**Reply by pythonweb2** [reply]
Although it looks like doing this breaks the model when sending the request via swagger-ui. So this is probably a bug.

**Comment by pdiegolopez** [comment, marked as answer]
It works with both Postman and Swagger using this example. Please note that I need to change the default media type. If you leave the default media type, it won’t work in Swagger, but it will work in Postman if you send it as Form Data (instead of x-www-urlencoded). I believe the FastAPI documentation should include this scenario.

```python
from fastapi import FastAPI, Form, UploadFile
from pydantic import BaseModel
from typing import Annotated, Optional


app = FastAPI()


class Info(BaseModel):
    text: str
    number: float
    image: Optional[UploadFile] = None


@app.post("/login/")
async def login(data: Annotated[Info, Form(media_type="multipart/form-data")]):
    if data.image:
        print(data.image.filename)
    return {"text": data.text, "number": data.number}

```

**Reply by AnRaTheStar17ju1682** [reply]
thanks a lot bro

**Comment by BrentHuang** [comment]
I'm having the same problem.

```
async def upload_print_files_(
                              files: Annotated[list[UploadFile],
                                               File(description="Multiple files as UploadFile")],
                              group_id: Annotated[int | None, Form(gt=0)] = None):
```

![image](https://github.com/user-attachments/assets/54f8a2e4-551a-4d80-b760-f7c8c6b5d955)

![image](https://github.com/user-attachments/assets/884a6e8f-1777-4d06-a2f3-82b0112544c4)

fastapi Version: 0.115.6

**Comment by LincolnPuzey** [comment]
@BrentHuang Even though you are using `Form(...)`, you are not using a [Form Model](https://fastapi.tiangolo.com/tutorial/request-form-models/), which is what this discussion is about.

The problem in your example is you are POSTing an empty string (`"input": ""`) to the `group_id` field, which can't be parsed to an `int` - hence the `"unable to parse string as integer"` error message.

**Comment by BrentHuang** [comment]
@LincolnPuzey Please note: I didn't enter any values manually, I just manipulated them on swaggerui. 

When I clear the group_id field on swaggerui, there should be no group_id field in the request that swaggerui sends to fastapi, instead of `group_id: ""`
